### PR TITLE
Aj export csv

### LINKED ISSUE #101: [Feature] Implement Export CSV Functionality
TODO

- [ ] A user should be able to click the export button to export the current table as a CSV file and save it to their machine
- [ ] This is with the same dummy data that is in the table already

Review by @mike10911 @kzeng24 @Aaryan1203

### PR DESCRIPTION
**Pull request by jmsargent10**
Fixes #101

- Wrapped the existing “Export CSV” button with something called a CSVLink which allows you to export a react object as a CSV file, with given data and headers, when clicked. You can read more about it [here](https://www.npmjs.com/package/react-csv).

- Changed cabinValues to be a list of lists instead of a dictionary so that it could comply with the CSVLink expected format of either an array of arrays, or an array of objects.
   - This change affected selectedCabin.tsx because before its useState was a string (a key from the cabinValues dict), but we needed it to be a number so we could index the list. We confirmed it still works as expected.

- Added two forEach loops to rearrange the emails in the exported CSV, as they were originally being displayed with the row and column positions swapped.

<br>

**When you click the “Export CSV” Button:**

<br>

**What the CSV file looks like in raw format:**

<br>

**What the CSV file looks like in table form:**

<br>

> N.B. One slight limitation is that there doesn’t seem to be a way to change the name of the file being downloaded…

### CODE REVIEW

**Comment by kzeng24** [comment]
Extra comments:
1. The cabin sorting link in the internal dashboard is missing - I think it was accidentally removed during merging.
2. When I accessed your branch, inside CSVCabinTable.tsx, it said that StyledTableHeader is missing so make sure to commit the StyledTableHeader in common.ts

**Review by mike10911 — CHANGES_REQUESTED** [review]
Awesome work guys! I'm also getting the issue of seeing HTML in the csv. It's possible that when addressing merge conflicts/rebasing, you might have undone that fix. 

P.S. don't worry about the file name issue, that feature is waiting for merge conflicts to be resolved to be merged in the react-csv repo. We can update this later!
https://github.com/react-csv/react-csv/pull/137

File: ui/components/templateDropdown/selectedCabin.tsx [hunk 1]
```diff
@@ -37,7 +38,7 @@ export default function SelectedCabin ({ cabinNames, cabinValues }: SelectedCabi
       <StyledFormControl style={{ width: '9em', display: 'inline-block' }}>
         <InputLabel>Cabin</InputLabel>
         <Select
-          value={selectedItem}
+          value={cabinNames[selectedItem]}
```
[the remark this answers is not in the document]
Comment (Mandy-cyber) [reply ui/components/templateDropdown/selectedCabin.tsx:41]: With the previous implementation, selectedItem was a string and was representative of a key in the cabinValues dictionary. But now that cabinValues is no loner a dict, and instead an array of arrays, we have to use integer indexing. Hope that answers that? Idk.

File: yarn.lock [hunk 2]
```diff
@@ -297,6 +297,16 @@ loose-envify@^1.1.0:
   dependencies:
     js-tokens "^3.0.0 || ^4.0.0"
 
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (Mandy-cyber) [reply yarn.lock:300]: good catch, thanks!

**Review by Aaryan1203 — APPROVED** [review]
Great work guys! I didn't find anything that needed any edits apart from a couple of the previous comments and I think you should be good to merge!

**Review by kzeng24 — COMMENTED** [review]
I still get this when I click on Export CSV :0
Screen Shot 2023-07-15 at 9 01 36 PM

**Review by mike10911 — APPROVED** [review]
LGTM!

**Review by kzeng24 — APPROVED** [review]
Approved! Nice job guys!!

**Pull request merged**